src/client/csfle/state_machine.rs

```diff
@@ -116,7 +116,7 @@ impl Client {
                             )
                             .await?;
                             stream.write_all(kms_ctx.message()?).await?;
-                            let mut buf = vec![];
+                            let mut buf = vec![0];
```

Comment:
why did clippy suggest this change? i can't seem to reproduce it locally 🤔

Reply:
Whoops, meant to comment on this.  This prevented a clippy false-positive below where it thought the read was into a zero-sized vec.